lib/new_relic/agent/system_info.rb

```diff
@@ -81,7 +81,7 @@ def self.get_processor_info
 
         @processor_info
       rescue
-        {}
+        NewRelic::EMPTY_HASH
```

Comment:
It looks like this one replaces a mutable hash with a frozen one, but from looking at the consumption of the return value, that should be totally fine. 👍 

The `get_processor_info` method itself is looking pretty long and gnarly though. If you'd ever like to take a crack at modernizing its approach to memoization and also breaking it up into several smaller methods (one per OS seems appropriate here), we can pair on doing so.

Reply:
Yes from what I could tell, we are only reporting on that hash in the case of an error or unknown processor. This was a crazy long method that I needed Tanna to go through with me, so I'd love to pair on cleaning this up!